src/vlt/test/commands/token.ts

```diff
@@ -0,0 +1,53 @@
+import { type Token } from '@vltpkg/registry-client'
+import t from 'tap'
+import { type LoadedConfig } from '../../src/types.js'
+
+const log: string[][] = []
+const { usage, command } = await t.mockImport<
+  typeof import('../../src/commands/token.js')
+>('../../src/commands/token.js', {
+  '@vltpkg/registry-client': {
+    async setToken(reg: string, tok: Token) {
+      log.push(['add', reg, tok])
+    },
+    async deleteToken(reg: string) {
+      log.push(['delete', reg])
+    },
+  },
+  '../../src/read-password.js': {
+    async readPassword(prompt: string) {
+      log.push(['readPassword', prompt])
+      return 'result'
+    },
+  },
+})
+
+t.matchSnapshot(usage().usageMarkdown())
+
+await command({
+  options: { registry: 'https://registry.vlt.javascript/' },
+  positionals: ['add'],
+} as unknown as LoadedConfig)
+
+await command({
+  options: { registry: 'https://registry.vlt.javascript/' },
+  positionals: ['rm'],
+} as unknown as LoadedConfig)
+
+t.strictSame(log, [
+  ['readPassword', 'Paste bearer token: '],
+  ['add', 'https://registry.vlt.javascript', 'Bearer result'],
+  ['delete', 'https://registry.vlt.javascript'],
+])
+
+t.test('invalid token sub command', async t => {
+  const exitCode = process.exitCode
+  const logs = t.capture(console, 'error').args
+  await command({
+    options: { registry: 'https://registry.vlt.javascript/' },
+    positionals: ['wat'],
+  } as unknown as LoadedConfig)
+  t.equal(process.exitCode, 1)
+  process.exitCode = exitCode
+  t.strictSame(logs(), [[usage().usage()]])
+})
```

Comment:
if the command is changed to throw `EUSAGE` instead of log directly, the `mockCommandOutput` helper can be used here which will run the command through the output helper and return an object of `{ errs, logs }` where `errs` will be everything logged to `stderr`

Reply:
or it can just assert that it throws. but i found it helpful doing the other commands to be able to test more integration style and get what would be output to the terminal

Reply:
Yeah, throwing an EUSAGE and asserting it now.